bfe.json

```diff
@@ -2,52 +2,55 @@
   {
     "code": 0,
     "type": "feed",
+    "sigil": "@",
     "formats": [
-      { "code": 0, "format": "ssb/classic",     "data_length": 32, "sigil": "@", "suffix": ".ed25519"   },
-      { "code": 1, "format": "ssb/gabby-grove", "data_length": 32, "sigil": "@", "suffix": ".ggfeed-v1" },
-      { "code": 2, "format": "bamboo",          "data_length": 32, "sigil": "@", "suffix": ".bamboo" },
-      { "code": 3, "format": "ssb/bendy-butt",  "data_length": 32, "sigil": "@", "suffix": ".bbfeed-v1" },
-      { "code": 4, "format": "ssb/fusion-identity", "data_length": 32, "sigil": "@", "suffix": ".fusion-v1" }
+      { "code": 0, "format": "ssb/classic",     "data_length": 32, "suffix": ".ed25519"   },
+      { "code": 1, "format": "ssb/gabby-grove", "data_length": 32, "suffix": ".ggfeed-v1" },
+      { "code": 2, "format": "bamboo",          "data_length": 32, "suffix": ".bamboo" },
+      { "code": 3, "format": "ssb/bendy-butt",  "data_length": 32, "suffix": ".bbfeed-v1" },
+      { "code": 4, "format": "ssb/fusion-identity", "data_length": 32, "suffix": ".fusion-v1" }
     ]
   },
   {
     "code": 1,
     "type": "msg",
+    "sigil": "%",
     "formats": [
-      { "code": 0, "format": "ssb/classic",     "data_length": 32, "sigil": "%", "suffix": ".sha256"   },
-      { "code": 1, "format": "ssb/gabby-grove", "data_length": 32, "sigil": "%", "suffix": ".ggmsg-v1" },
-      { "code": 2, "format": "",                "data_length": 32, "sigil": "%", "suffix": ".cloaked" },
-      { "code": 3, "format": "bamboo",          "data_length": 64, "sigil": "%", "suffix": ".bamboo" },
-      { "code": 4, "format": "ssb/bendy-butt",  "data_length": 32, "sigil": "%", "suffix": ".bbmsg-v1" }
+      { "code": 0, "format": "ssb/classic",     "data_length": 32, "suffix": ".sha256"   },
+      { "code": 1, "format": "ssb/gabby-grove", "data_length": 32, "suffix": ".ggmsg-v1" },
+      { "code": 2, "format": "cloaked msg",     "data_length": 32, "suffix": ".cloaked" },
+      { "code": 3, "format": "bamboo",          "data_length": 64, "suffix": ".bamboo" },
+      { "code": 4, "format": "ssb/bendy-butt",  "data_length": 32, "suffix": ".bbmsg-v1" }
     ]
   },
   {
     "code": 2,
     "type": "blob",
+    "sigil": "&",
     "formats": [
-      { "code": 0, "format": "ssb/classic", "data_length": 32, "sigil": "&", "suffix": ".sha256" }
+      { "code": 0, "format": "ssb/classic", "data_length": 32, "suffix": ".sha256" }
     ]
   },
   {
     "code": 3,
     "type": "diffie-hellman",
     "formats": [
-      { "code": 0, "format": "curve25519", "key_length": 32 }
+      { "code": 0, "format": "curve25519", "data_length": 32, "key_length": 32 }
     ]
   },
   {
     "code": 4,
     "type": "signature",
     "formats": [
-      { "code": 0, "format": "ed25519", "signature_length": 64 }
+      { "code": 0, "format": "ed25519", "data_length": 64, "signature_length": 64, "suffix": ".sig.ed25519" }
```

Comment:
I stopped short or removing unique `X_length` keys, as this would be a breaking change.
would like to see them go though

Reply:
I mean I think the key should always be data_length